internal/resources/product/model.go

```diff
@@ -0,0 +1,470 @@
+package product
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/elliotchance/pie/v2"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/labd/commercetools-go-sdk/platform"
+	"github.com/labd/terraform-provider-commercetools/internal/customtypes"
+	"github.com/labd/terraform-provider-commercetools/internal/utils"
+)
+
+// Product represents the main schema data.
+type Product struct {
```

Comment:
I am not 100% sure about this data model. The actual implementation for a product has the `masterData` field, which contains both the current and staged versions of the product. I am uncertain if we want to mirror this data model in the provider directly, or if we want to reduce the complexity a bit by making the assumption that a product that is managed in terraform should always be in a published state (so we can basically ignore staged, as it will always be the same as published). What is your usecase here?

Reply:
I have implemented the approach you described where the stage and current values are always the same.
Personally, I do not see the need to manage these values separately in the Terraform template.